lib/client/debug/debug.go

```diff
@@ -137,6 +142,27 @@ func (c *Client) CollectProfile(ctx context.Context, profileName string, seconds
 	return result, nil
 }
 
+// GetReadiness returns true if the Teleport service is ready.
+func (c *Client) GetReadiness(ctx context.Context) (ready bool, msg string, err error) {
+	resp, err := c.do(ctx, http.MethodGet, url.URL{Path: "/readyz"}, nil)
+	if err != nil {
+		return false, " ", trace.Wrap(err)
+	}
```

Comment:
We could get a dial error if an old socket is left on disk (but nothing listens on it, so it's not connectable) and the new process hasn't replaced it yet.